sdks/go/pkg/beam/io/webapi/doc.go

```diff
@@ -0,0 +1,55 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package webapi supports reading from or writing to Web APIs.
```

Comment:
I know this is a work in progress, but the end goal isn't yet clear from this PR. That is the context of the following comment.

Right now, this is (at present) an alternative API wrapper for a DoFn, that has a dead letter queue. So, as presently provided, how different is the best future experience this API provides vs a user writing a DoFn directly? What does that future experience & benefits look like?

Remember, the point of an API wrapper like this is to make it easier for a user to do a complicated thing vs directly doing that complicated thing in the framework. Right now, it's about the same. But with throttling, retries, etc, or similar that are harder to implement right, but could be made convenient and consistent... that changes quickly.